doc/rest-api.md

```diff
@@ -2631,6 +2633,79 @@ Input (none at present):
     {
     }
 
+
+## `/1.0/storage-pools/<pool>/volumes/<type>/<name>/snapshots`
+### GET
+ * Description: List of volume snapshots
+ * Authentication: trusted
+ * Operation: sync
+ * Return: list of URLs for snapshots for this volume
+
+Return value:
+
+    [
+        "/1.0/storage-pools/default/volumes/custom/foo/snapshots/snap0"
+    ]
+
+### POST
+ * Description: create a new volume snapshot
+ * Authentication: trusted
+ * Operation: async
+ * Return: background operation or standard error
+
+Input:
+
+    {
+        "name": "my-snapshot",          # Name of the snapshot
+    }
+
+## `/1.0/storage-pools/<pool>/volumes/<type>/<volume>/snapshots/name`
+### GET
+ * Description: Snapshot information
+ * Authentication: trusted
+ * Operation: sync
+ * Return: dict representing the snapshot
+
+Return:
+
+    {
+        "config": {},
+        "description": "",
+        "name": "snap0"
+    }
+
+### PUT
+ * Description: Volume snapshot information
+ * Authentication: trusted
+ * Operation: sync
```

Comment:
So the documentation agrees that this is a sync function, it's just the implementation that doesn't :)